test/jdk/java/awt/TrayIcon/DisposeInActionEventTest/DisposeInActionEventTest.java

```diff
@@ -21,76 +21,76 @@
-                    "Simply press PASS button."
-            };
-        } else {
-            String clickInstruction;
-            if (Platform.isOSX()) {
-                clickInstruction = "right";
-            } else {
-                clickInstruction = "left";
-            }
-            instructions = new String[]{
-                    "When the test starts, it adds the icon to the tray aread. If you",
-                    "  don't see a tray icon, please, make sure that the tray area",
-                    "  (also called Taskbar Status Area on MS Windows, Notification",
-                    "  Area on Gnome or System Tray on KDE) is visible.",
-                    "Double-click with " + clickInstruction + " button on the tray icon to trigger the",
-                    "  action event. Brief information about action events is printed",
-                    "  below. After each action event the tray icon is removed from",
-                    "  the tray and then added back in a second.",
-                    "The test performs some automatic checks when removing the icon. If",
-                    "  something is wrong the corresponding message is displayed below.",
-                    "  Repeat double-clicks several times. If no 'Test FAILED' messages",
-                    "  are printed, press PASS button else FAIL button."
-            };
+        String instructions = "When the test starts, it adds the icon to the tray area. If you\n" +
+                       "  don't see a tray icon, please, make sure that the tray area\n" +
+                       "  (also called Taskbar Status Area on MS Windows, Notification\n" +
+                       "  Area on Gnome or System Tray on KDE) is visible.\n\n" +
+                        clickInstruction + " the button on the tray icon to trigger the\n" +
+                       "  action event. Brief information about action events is printed\n" +
+                       "  in the frame. After each action event the tray icon is removed from\n" +
+                       "  the tray and then added back in a second.\n\n" +
+                       "The test performs some automatic checks when removing the icon. If\n" +
+                       "  something is wrong the corresponding message the test will automatically fail.\n" +
+                       "  Repeat clicks several times. If no 'Test FAILED' messages\n" +
+                       "  are printed, press PASS button else FAIL button.";
+
+        try {
+            PassFailJFrame.builder()
+                    .title("DisposeInActionEventTest")
+                    .instructions(instructions)
+                    .testTimeOut(10)
+                    .rows(15)
+                    .columns(45)
+                    .testUI(DisposeInActionEventTest::showFrameAndIcon)
+                    .build()
+                    .awaitAndCheck();
+        } finally {
+            systemTray.remove(trayIcon);
         }
```

Comment:
@alisenchung I missed adding null check here earlier. It is good to have this safety check to avoid NPE.

```suggestion
        } finally {
            if (systemTray != null) {
               systemTray.remove(trayIcon);
            }
         }
```

Reply:
`SystemTray.getSystemTray()` can't return `null`. However, the static field `systemTray` can remain `null` in the test.